build-system/tasks/dep-check.js

```diff
@@ -180,35 +180,15 @@ async function getEntryPointModule() {
  * @return {!ModuleDef}
  */
 async function getModuleGraph(entryPointModule) {
-  const esbuildBabelPlugin = {
-    name: 'babel',
-    async setup(build) {
-      const transformContents = async ({file, contents}) => {
-        const babelOptions = babel.loadOptions({
-          caller: {name: 'dep-check'},
-          filename: file.path,
-          sourceFileName: path.relative(process.cwd(), file.path),
-        });
-        const result = await babel.transformAsync(contents, babelOptions);
-        return {contents: result.code};
-      };
-
-      build.onLoad({filter: /.*/, namespace: ''}, async (file) => {
-        const contents = await fs.promises.readFile(file.path, 'utf-8');
-        const transformed = await transformContents({file, contents});
-        return transformed;
-      });
-    },
-  };
-
   const bundleFile = path.join(depCheckDir, 'entry-point-bundle.js');
   const moduleGraphFile = path.join(depCheckDir, 'module-graph.json');
+  const plugin = getEsbuildBabelPlugin('dep-check', /* enableCache */ false);
```

Comment:
note: once this is through, we should be able to trim many babel plugins. using esbuild's equivalent (e.g. for jsx) should be much faster, as long as we don't require es5 transpilation (and I wouldn't expect dep-check needs much)